dev/src/index.ts

```diff
@@ -1303,17 +1395,12 @@ export class Firestore implements firestore.Firestore {
       });
     };
 
-    docStream
-      .on('error', err => {
-        err.code = Status.UNAVAILABLE;
-        err.stack = 'Failed to fetch children documents: ' + err.stack;
-        lastError = err;
-        onStreamEnd();
-      })
-      .on('data', (snap: QueryDocumentSnapshot) => {
-        writer.delete(snap.ref).catch(err => incrementErrorCount(err));
-      })
-      .on('end', () => onStreamEnd());
+    const docStream = this._getAllDescendants(
+      ref instanceof CollectionReference
+        ? (ref as CollectionReference<unknown>)
+        : (ref as DocumentReference<unknown>)
+    );
```

Comment:
I think this could be done in `setupStream` to remove some duplication. I think we should also pass the limit here as it is otherwise not clear where it comes from when looking at the callsite.

Reply:
Done. I'm passing the limit in `getAllDescendants()` now.